apiserver/facades/client/controller/controller.go

```diff
@@ -1029,22 +1048,17 @@ func makeModelInfo(ctx context.Context, st, ctlrSt *state.State,
 	}
 
 	// Retrieve agent version for the model.
-	conf, err := model.ModelConfig(ctx)
+	modelID := coremodel.UUID(model.UUID())
+	modelInfo, err := modelService.Model(ctx, modelID)
 	if err != nil {
-		return empty, userList{}, errors.Trace(err)
+		return empty, userList{}, fmt.Errorf("getting model %q: %w", modelID, err)
 	}
-	agentVersion, _ := conf.AgentVersion()
 
 	// Retrieve agent version for the controller.
-	controllerModel, err := ctlrSt.Model()
-	if err != nil {
-		return empty, userList{}, errors.Trace(err)
-	}
-	controllerConfig, err := controllerModel.Config()
+	controllerModel, err := modelService.ControllerModel(ctx)
 	if err != nil {
-		return empty, userList{}, errors.Trace(err)
+		return empty, userList{}, fmt.Errorf("getting agent version for controller model: %w", err)
```

Comment:
issue: error message is misleading. the failure is in getting the controller model info.